### PR TITLE
Update single page#111

### PR DESCRIPTION
**Pull request by brenmeyer2245**
-Moved Links for Edit and Start Event into Single Page 

-Retooled Player-add and Player-List to fetch and update an array of Users Associated with the selected event

-Renamed reducer and most of the items associated with it To UsersAtEvent ie All the Users at this Event 

-Commented out the front end tests “players” while I retooled 

-Added List of Users attending event and Add Users for Single Event Page 

-Added a Loading option for Single event page in case the component renders before mapState gets the event details. 
Had trouble with this and will continue to look into it this weekend. 

-Added Api routes for Put and Fetch from users at events 

-Created Middleware mounted on Users users/atEvents to keep things somewhat RESTful

### CODE REVIEW

File: client/components/game-forms/EventForm.js [hunk 1]
```diff
@@ -1,38 +1,39 @@
-/*global M */
```
Comment (kevslim) [inline client/components/game-forms/EventForm.js]: Don't we still need the global M or do we not use this EventForm in our edit event page

File: client/store/player.js [hunk 2]
```diff
@@ -1,56 +1,54 @@
...
-export const deletePlayer = playerId => dispatch => {
-  //using playerId as the email name since the temp db doesn't have an id
-  dispatch(removePlayer(playerId))
-}
-export const createPlayer = player => dispatch => {
-  const {email} = player
-  dispatch(addPlayer(email))
+
+export const fetchPlayersByEventId = eventId => async dispatch => {
+  const {data: players} = await axios.get(`/api/users/atEvents/${eventId}`)
```
Comment (pckilgore) [inline client/store/player.js]: Ideally try/catch here.

File: client/store/player.js [hunk 3]
```diff
@@ -1,56 +1,54 @@
...
+  dispatch(gotUsersForEvent(players))
 }
-export const fetchAllPlayers = () => dispatch => {
-  dispatch(getPlayers(db))
+
+export const addUserToEvent = (userEmail, eventId) => async dispatch => {
+  const {data: user} = await axios.put('/api/users/atEvents', {
```
Comment (pckilgore) [inline client/store/player.js]: Probably do try/catches here.

File: server/api/usersAtEvents.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+const {User, Event} = require('../db/models')
+const canOnlyBeUsedBy = require('./authMiddleware')
+module.exports = router
+
+router.get(
+  '/:eventId',
+  canOnlyBeUsedBy('admin', 'leader', 'self', 'participant'),
+  async (req, res, next) => {
+    try {
+      const events = await Event.findById(req.params.eventId, {
```
Comment (pckilgore) [inline server/api/usersAtEvents.js]: We just talked about this.  It works, but feels weird. You should be able to make a more direct query.
```
Event
  .findById(...)
  .then(event=>event.getUsers())
  .then(users=>res.json(users))
  .catch(next)
```

**Pull request merged**